diff -rcNB tmp2/runtime/src/support/Any.h tmp/runtime/src/support/Any.h
*** tmp2/runtime/src/support/Any.h	2017-07-26 11:10:02.000000000 -0700
--- tmp/runtime/src/support/Any.h	2018-04-20 09:04:19.082012100 -0700
***************
*** 99,105 ****
    }
  
  private:
!   struct Base {
      virtual ~Base();
      virtual Base* clone() const = 0;
    };
--- 99,105 ----
    }
  
  private:
!   struct ANTLR4CPP_PUBLIC Base {
      virtual ~Base();
      virtual Base* clone() const = 0;
    };
